src/Microsoft.AspNetCore.Mvc.ViewFeatures/Resources.resx

```diff
@@ -292,4 +292,7 @@
   <data name="TempDataProperties_PublicGetterSetter" xml:space="preserve">
     <value>The '{0}.{1}' property with {2} is invalid. A property using {2} must have a public getter and setter.</value>
   </data>
+  <data name="ModelExpression_InvalidProperty" xml:space="preserve">
+    <value>Could not create a {0} for property '{1}'. Make sure the property is public.</value>
```

Comment:
Does this suggestion resolve all of your other feedback around the error message?